pkg/network/route_cache.go

```diff
@@ -275,14 +275,19 @@ func (n *netlinkRouter) Route(source, dest util.Address, netns uint32) (Route, b
 		})
 
 	if err != nil {
-		routeCacheTelemetry.netlinkErrors.Inc()
 		if iifIndex > 0 {
-			if errno, ok := err.(syscall.Errno); ok && (errno == syscall.EINVAL || errno == syscall.ENODEV) {
-				// invalidate interface cache entry as this may have been the cause of the netlink error
-				n.removeInterface(source, netns)
+			errno, ok := err.(syscall.Errno)
+			if ok {
+				netlinkErrIncWithTag(errno)
+				if errno == syscall.EINVAL || errno == syscall.ENODEV {
+					// invalidate interface cache entry as this may have been the cause of the netlink error
+					n.removeInterface(source, netns)
+				}
 			}
 		}
+		log.Debugf("Error getting route via netlink from %s: %s", dstIP, err)
 	} else if len(routes) != 1 {
+		log.Debugf("Failed to get route from cache for %s", dstIP)
```

Comment:
Can we make this more specific, like `Did not exactly one route for %s, got %d routes`?